src/traffic_server/traffic_server.cc

```diff
@@ -217,6 +235,7 @@ static ArgumentDescription argument_descriptions[] = {
   {"bind_stderr", '-', "Regular file to bind stderr to", "S512", &bind_stderr, "PROXY_BIND_STDERR", nullptr},
   {"accept_mss", '-', "MSS for client connections", "I", &accept_mss, nullptr, nullptr},
   {"poll_timeout", 't', "poll timeout in milliseconds", "I", &poll_timeout, nullptr, nullptr},
+  {"block", '-', "block for debug attach", "T", &cmd_block, nullptr, nullptr},
```

Comment:
This looks promising.

I suggest we update the MakeATSProcess trafficserver extension in autest-site to take a flag that configures this for the user. Maybe `block_for_detach=False` and someone wanting to use this can pass `block_for_detatch=True` and this will set things up for that, doing the necessary to make things work for cmd being traffic_server or traffic_manager.